shared/engine/index.js

```diff
@@ -161,20 +161,42 @@ class Engine {
   }
 
   // An outgoing call. ONLY called by the flow-type rpc helpers
-  _rpcOutgoing (params: {
-    method: MethodKey,
-    param?: ?Object,
-    incomingCallMap?: incomingCallMapType,
-    callback?: ?(...args: Array<any>) => void,
-    waitingHandler?: WaitingHandlerType}
+  _rpcOutgoing (
+    method: string,
```

Comment:
instead of one object we accept the method, params and overrides to callback and incomingCallMap which the ChannelMap uses